modules/functions.php

```diff
@@ -103,6 +103,32 @@ function getSystemLoadAvg()
   return sys_getloadavg ()[2];
 }
 
+// get system memory info
+function getSystemMemInfo() 
+{       
+    $data = explode("\n", file_get_contents("/proc/meminfo"));
+    $meminfo = array();
+    foreach ($data as $line) {
+        list($key, $val) = explode(":", $line);
+        $meminfo[$key] = trim($val);
+    }
+    return $meminfo;
+}
+
+// get system total memory in MB
+function getSystemTotalMem()
+{
+    return intval(getSystemMemInfo()["MemTotal"] / 1024);
+}
+
+// get system used memory in MB
+function getSystemAvailMem()
+{
+    $meminfo = getSystemMemInfo();
+    return intval(($meminfo["MemTotal"] - $meminfo["MemAvailable"]) / 1024);
```

Comment:
I think it should be $meminfo["MemTotal"] - $meminfo["MemFree"].

Reply:
Nope. Linux uses the ram as cache. So on machines that are running some time there will only be a few hundred MB "free". But the ram used for caching will be freed the moment a process needs it. MemFree is only the really "free" ram. But MemAvailable is the "free" and the cached ram. So it is more meaningfull than MemFree.

Reply:
With "MemAvailable" my 1GB node shows a usage of 302 MB / 997 MB while "htop" reports 53.7% MEM. According to the author of htop, "Total Used Memory" is calculated as Total - Free in htop, see https://stackoverflow.com/questions/41224738/how-to-calculate-system-memory-usage-from-proc-meminfo-like-htop

Reply:
Yes that is right but MemFree does not really tell the truth as there is no problem using the cached but not free memory if it is needed. Check here: https://superuser.com/a/980821

Reply:
What about using the same metrics as "free", see 'man free'?

free - Display amount of free and used memory in the system

       total  Total installed memory (MemTotal and SwapTotal in /proc/meminfo)

       used   Used memory (calculated as total - free - buffers - cache)

       free   Unused memory (MemFree and SwapFree in /proc/meminfo)

       shared Memory used (mostly) by tmpfs (Shmem in /proc/meminfo, available
              on kernels 2.6.32, displayed as zero if not available)

       buffers
              Memory used by kernel buffers (Buffers in /proc/meminfo)

       cache  Memory  used  by  the  page  cache and slabs (Cached and Slab in
              /proc/meminfo)

       buff/cache
              Sum of buffers and cache

       available
              Estimation of how much memory  is  available  for  starting  new
              applications,  without swapping. Unlike the data provided by the
              cache or free fields, this field takes into account  page  cache
              and also that not all reclaimable memory slabs will be reclaimed
              due to items being in use (MemAvailable in /proc/meminfo, avail‐
              able on kernels 3.14, emulated on kernels 2.6.27+, otherwise the
              same as free)

Reply:
It is the same as before. The available ram is the interesting thing. On one of my nodes free is 60MB where I would be really worried about the system not having enough ram. The available memory is over 700MB though. If you really want I can make a commit to make it configurable but I don't think the MemFree is relevant.

Reply:
I guess it's OK. Thanks :)